games/rummy/scenes/PlayScene.ts

```diff
@@ -0,0 +1,1160 @@
+      zoneX + 127,
+      zoneY - 95,
+      GAME.CARD.WIDTH,
+      GAME.CARD.HEIGHT
+    );
+
+    this.backDeckZone = this.add.zone(
+      this.deckBackCardX - GAME.CARD.WIDTH / 2,
+      this.deckBackCardY - GAME.CARD.HEIGHT / 2,
+      GAME.CARD.WIDTH,
+      GAME.CARD.HEIGHT
+    );
+
+    // ゾーンの可視化用のグラフィックスオブジェクトを作成
+    const graphics = this.add.graphics();
+    const graphics2 = this.add.graphics();
+
+    // ゾーンの位置とサイズを表す長方形を描画
+    graphics.lineStyle(2, 0xff0000);
+    graphics.strokeRect(
+      zoneX + 127,
+      zoneY - 95,
+      GAME.CARD.WIDTH,
+      GAME.CARD.HEIGHT
+    );
+    this.add.existing(graphics);
+    this.frontDeckZone.setInteractive();
+    this.frontDeckZone.on('pointerdown', () => {
+      console.log('Zone clicked!');
+    });
+
+    graphics2.lineStyle(2, 0xff0000);
+    graphics2.strokeRect(
+      this.deckBackCardX - GAME.CARD.WIDTH / 2,
+      this.deckBackCardY - GAME.CARD.HEIGHT / 2,
+      GAME.CARD.WIDTH,
+      GAME.CARD.HEIGHT
+    );
+    this.add.existing(graphics2);
+    this.backDeckZone.setInteractive();
+    this.backDeckZone.on('pointerdown', () => {
+      console.log('Zone2 clicked!');
+    });
+  }
+
+  private createDeck() {
+    this.deckBack = [
+      new Deck(
+        this,
+        this.deckBackCardX,
+        this.deckBackCardY,
+        ['Hearts', 'Diamonds', 'Spades', 'Clubs']
+      )
+    ];
+    this.deckBack?.forEach((deck) => {
+      deck.shuffle();
+      console.log(deck);
+      const curr = deck.drawOne();
+      console.log(deck);
+      if (curr) this.deckFront.push(curr);
+      this.deckFront.forEach((card) => {
+        card.playMoveTween(
+          this.deckFrontCardX,
+          this.deckFrontCardY
+        );
+        card.playFlipOverTween();
+      });
+    });
+  }
+
+  private descriptionFromDeckToHandOfPlayerAndHouse() {
+    this.players.forEach((player, index) => {
+      this.deckBack?.forEach((deck) => {
+        if (player.playerType === 'player') {
+          this.createHands(deck, index, player);
+        } else {
+          this.createHands(deck, index, player);
+        }
+      });
+    });
+  }
+
+  // game starting
+  private async startGame() {
+    await this.delay(4000);
+    if (
+      this.players[0].hand.length < 1 ||
+      this.players[1].hand.length < 1
+    ) {
+      console.log('ゲーム終了');
+      await this.showTurnLog();
+    }
+
+    if (this.isPlayerTurn) {
+      console.log('player Turn !!!');
+      console.log(this.arrayCardsPutOnMeldZone);
+      await this.showTurnLog();
+
+      await this.playerTurn().then(() => {
+        this.isPlayerTurn = !this.isPlayerTurn;
+        this.startGame();
+      });
+    } else {
+      console.log('house Turn !!!');
+      console.log(this.arrayCardsPutOnMeldZone);
+      await this.showTurnLog();
+
+      await this.houseTurn().then(() => {
+        this.isPlayerTurn = !this.isPlayerTurn;
+        this.startGame();
+      });
+    }
+  }
+
+  // 手札作成関数
+  private createHands(
```

Comment:
dealInitialCardsとかの方が分かりやすいです。


```suggestion
  private dealInitialCards(
```